src/Microsoft.SqlTools.ServiceLayer/Rename/RenameUtils.cs

```diff
@@ -0,0 +1,79 @@
+        {
+            if (requestParams == null)
+            {
+                throw new ArgumentNullException();
+            }
+            if (String.IsNullOrEmpty(requestParams.TableInfo.TableName) || String.IsNullOrEmpty(requestParams.TableInfo.Database) || String.IsNullOrEmpty(requestParams.TableInfo.Schema) || String.IsNullOrEmpty(requestParams.TableInfo.OldName) || String.IsNullOrEmpty(requestParams.ChangeInfo.NewName))
+            {
+                throw new ArgumentException(SR.RenameRequestParametersNotNullOrEmpty);
+            }
+            if (!Regex.IsMatch(requestParams.TableInfo.Schema, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.Database, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.OldName, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.TableName, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.ChangeInfo.NewName, RegexMatchPatternValidObjectName))
+            {
+                throw new ArgumentOutOfRangeException(SR.NotAllowedCharacterForRenaming);
+            }
+        }
+        /// <summary>
+        /// Method to combine the schema, tablename and the columnname
+        /// </summary>
+        /// <param name="schema">schema of the database object</param>
+        /// <param name="tableName">tablename on which the renaming operation is executed</param>
+        /// <param name="column">column which should be renamed (optional)</param>
+        /// <returns>string which contains the three element separated by a dot</returns>
+        public static string CombineTableNameWithSchema(string schema, string tableName, string column = "")
+        {
+            schema = schema.Replace("[", "").Replace("]", "").Trim();
+            tableName = tableName.Replace("[", "").Replace("]", "").Trim();
+            column = column.Replace("[", "").Replace("]", "").Trim();
+            if (!String.IsNullOrEmpty(column))
+            {
+                return String.Join(".", schema, tableName, column);
+            }
+            return String.Join(".", schema, tableName);
+        }
+        /// <summary>
+        /// Method to generate the sql command to execute the renaming operation with the help of the sp_rename stored procedure
+        /// </summary>
+        /// <param name="requestParams">parameters which contains the information for the renaming operation</param>
+        /// <returns>the command to execute the renaming operation</returns>
+        public static string GetRenameSQLCommand(ProcessRenameEditRequestParams requestParams)
+        {
+            Logger.Verbose("Inside in the GetRenameSQLCommand()-Method");
+            if (requestParams.ChangeInfo.Type == ChangeType.COLUMN)
+            {
+                return String.Format(@"
+                USE [{0}];
+                    EXEC sp_rename @objname = '{1}', @newname = '{2}', @objtype ='{3}';
```

Comment:
Instead of directly defining the query here I wonder if we should instead be using SMO to handle that for us - that'd save us from duplicating any logic needed. 

@shueybubbles SSMS uses SMO for this, right? Is there any reason you can think of that it wouldn't be a good idea to do that here?

Reply:
I did not know that there is an extra project for this (SMO). Nonetheless, I am open if I should include this.   

The only reason I could think of for keeping this would be if at some point in Azure Data Studio it would be necessary to script the renaming operation. Through this you would have the script right away. However, I don't know if the SMO project can possibly do this already.